jenkins/docker/docker-re-release.jenkinsfile

```diff
@@ -0,0 +1,59 @@
+lib = library(identifier: 'jenkins@5.7.1', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+pipeline {
+    options {
+        timeout(time: 7, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+            name: 'PRODUCT',
+            description: "OpenSearch / OpenSearch-Dashboards",
+            trim: true
+        )
+        string(
+            name: 'TAG',
+            description: "Enter tag of the Product",
+            trim: true
+        )
+        booleanParam(
+            name: 'RE_RELEASE',
```

Comment:
Not sure the point of this param if the workflow itself is aimed for re-release already?
What is the point of rebuild but not re-release?
What does it force in this case?

Reply:
This parameter acts as a buildOption to create a docker-copy after the successful build inside the [buildDockerImage()](https://github.com/opensearch-project/opensearch-build-libraries/blob/560f671d486af6e78af1bb21bc7591c5388184d1/vars/buildDockerImage.groovy#L71) 
I will update the description.

Reply:
So in which case you do not want to docker-copy?
If not for docker-copy after we can just use docker-build without this workflow tho.
I am still not really sure why we need this at all.